crates/core/src/host/instance_env.rs

```diff
@@ -144,23 +145,24 @@ impl InstanceEnv {
     ///
     /// Returns an error if no columns were deleted or if the column wasn't found.
     #[tracing::instrument(skip(self, value))]
-    pub fn delete_by_col_eq(&self, table_id: u32, col_id: u32, value: &[u8]) -> Result<u32, NodesError> {
+    pub fn delete_by_col_eq(&self, table_id: u32, col_id: u32, value: &[u8]) -> Result<NonZeroU32, NodesError> {
         let stdb = &*self.dbic.relational_db;
         let tx = &mut *self.get_tx()?;
 
         // Interpret the `value` using the schema of the column.
         let eq_value = stdb.decode_column(tx, table_id, col_id, value)?;
 
         // Find all rows in the table where the column data equates to `value`.
-        let seek = stdb.iter_by_col_eq(tx, table_id, ColId(col_id), eq_value)?;
-        let seek = seek.map(|x| x.view().clone()).collect::<Vec<_>>();
+        let rows_to_delete = stdb
+            .iter_by_col_eq(tx, table_id, ColId(col_id), eq_value)?
+            .map(|x| *x.id())
+            .collect::<Vec<_>>();
 
         // Delete them and count how many we deleted and error if none.
         let count = stdb
-            .delete_by_rel(tx, table_id, seek)
-            .inspect_err_(|e| log::error!("delete_by_col_eq(table_id: {table_id}): {e}"))?
-            .ok_or(NodesError::ColumnValueNotFound)?;
-
+            .delete(tx, table_id, rows_to_delete)
```

Comment:
This seems like it will now make a call into LockingTxDatastore per row to delete. And therefore in particular take the lock over and over. Was that intentional?